packit/base_git.py

```diff
@@ -351,21 +351,40 @@ def fetch_upstream_archive(self):
         with cwd(self.absolute_specfile_dir):
             self.download_remote_sources()
 
-    def set_specfile_content(self, specfile: Specfile, version: str, comment: str):
+    def set_specfile_content(
+        self,
+        specfile: Specfile,
+        version: str,
+        comment: str,
+        preserve_autochangelog: bool = False,
+    ):
         """
         update this specfile using provided specfile
 
         :param specfile: specfile to get changes from (we update self.specfile)
         :param version: version to set in self.specfile
         :param comment: new comment for the version in %changelog
+        :param preserve_autochangelog: when it is set the %autochangelog macro is written
+                                       to the downstream spec and no changelog entry is added
         """
         previous_changelog = self.specfile.spec_content.section("%changelog")
         self.specfile.spec_content.sections[:] = specfile.spec_content.sections[:]
-        self.specfile.spec_content.replace_section(
-            "%changelog", previous_changelog or []
-        )
+        if preserve_autochangelog:
+            self.specfile.spec_content.replace_section(
+                "%changelog",
+                [
+                    "%autochangelog",
+                ],
+            )
+        else:
+            self.specfile.spec_content.replace_section(
+                "%changelog", previous_changelog or []
+            )
         self.specfile.save()
-        self.specfile.set_spec_version(version=version, changelog_entry=comment)
```

Comment:
to me, it would probably make sense to do it like:

```py
changelog_entry = None if preserve_autochangelog else comment
self.specfile.set_spec_version(version=version, changelog_entry=changelog_entry)
```

Reply:
I think I can not do it like this for the reason that I explained to @csomh

Reply:
The previous if-else block could also be refactored the same way, which I think would improve readability (although that's subjective :) ).